client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -20,71 +14,91 @@ import {
 	getTwoFactorRememberMe,
 	getTwoFactorPushPollInProgress
 } from 'state/login/selectors';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { rawHttp } from 'state/data-layer/http/actions';
 
 /***
  * Module constants
  */
 const POLL_APP_PUSH_INTERVAL_SECONDS = 5;
 
 /***
- * Checks the status of the push notification auth
+ * Dispatch action to perform http request
  *
- * @param {Object}   store  Global redux store
- * @returns {Promise}		Promise of result from the API
+ * @param {Object}	store  Global redux store
+ * @param {Object}	action dispathced action
  */
-const doAppPushRequest = ( store ) => {
-	return request.post( 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint' )
-		.withCredentials()
-		.set( 'Content-Type', 'application/x-www-form-urlencoded' )
-		.accept( 'application/json' )
-		.send( {
+const requestTwoFactorPushNotificationStatus = ( store, action ) => {
+	store.dispatch( rawHttp( {
+		url: 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint',
```

Comment:
it looks like maybe when this is all done we should move this file. it's not in the [right place in the data layer](https://github.com/Automattic/wp-calypso/tree/481899c7d7d6d8112cae943ffc5fb9fcab2968d8/client/state/data-layer#file-structure) based on the URL it's hitting whereas it should probably go into a new `http` directory or maybe inside the `third-party` directory - something like `client/state/data-layer/http/wordpress.com/wp-login.js`

this is to provide a mapping between actual requests going over the wire and where in the codebase to look to find them.